modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -535,6 +542,24 @@ private void updateLeaseBatchInternal() {
                 );
             }
 
+            leasesCurrent.leaseByGroupId().forEach(renewedLeases::putIfAbsent);
+
+            for (Iterator<Entry<ReplicationGroupId, Lease>> iter = renewedLeases.entrySet().iterator(); iter.hasNext(); ) {
+                Map.Entry<ReplicationGroupId, Lease> entry = iter.next();
+                ReplicationGroupId groupId = entry.getKey();
+                Lease lease = entry.getValue();
+
+                if (clockService.before(lease.getExpirationTime(), currentTime)
+                        && !groupsAmongCurrentStableAndPendingAssignments.contains(groupId)) {
+                    iter.remove();
+                } else if (prolongableLeaseGroupIds.contains(groupId)
+                        && !lease.getExpirationTime().equals(newExpirationTimestamp)) {
+                    entry.setValue(prolongLease(groupId, lease, null, newExpirationTimestamp));
+                }
+            }
+
+            byte[] renewedValue = new LeaseBatch(renewedLeases.values()).bytes();
+
             if (Arrays.equals(leasesCurrent.leasesBytes(), renewedValue)) {
```

Comment:
This condition has to be moved above the last loop.

Reply:
No, the meaning of this `if` is to remove excessive invokes. If the final renewed value differs from leasesCurrent, then invoke makes sense, otherwise not.
It should be fixed in another way.

Reply:
I moved it and modified the condition.